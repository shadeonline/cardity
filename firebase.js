// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword} from 'firebase/auth';
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCf2BsSO7intMEZyZQUbf3gdQK_TNOvxHE",
  authDomain: "carditydb.firebaseapp.com",
  projectId: "carditydb",
  storageBucket: "carditydb.appspot.com",
  messagingSenderId: "627053711755",
  appId: "1:627053711755:web:caa91da4f88ef749419f8e",
  measurementId: "G-44WRTNPMRV"
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const auth = getAuth(app);

export { auth};