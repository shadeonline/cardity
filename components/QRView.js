import React from 'react';
import { Text, View } from 'react-native';
import QRCode from 'react-native-qrcode-svg';
import { auth } from '../firebase';

const QRView = ({ card }) => {
    const userUID = auth.currentUser?.uid;
    const cardDetails = 'cardity:'+JSON.stringify({ ...card, userUID });
    console.log(cardDetails);

    return (
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', backgroundColor: 'white' }}>
            <Text>{card.cardName}</Text>
            <QRCode
                value={cardDetails}
                size={200}
                color="black"
                backgroundColor="white"
            />
        </View>
    );
};

export default QRView;
