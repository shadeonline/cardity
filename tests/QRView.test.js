import React from 'react';
import { render } from 'react-native-testing-library';
import QRView from '../components/QRView';
import QRCode from 'react-native-qrcode-svg';
import { auth } from '../firebase';

jest.mock('react-native-qrcode-svg', () => {
    // Replace 'QRCode' with the name of the component you're mocking
    const MockedQRCode = () => null;
    return MockedQRCode;
  });

  jest.mock('../firebase', () => ({
    auth: {
      currentUser: {
        uid: 'mockedUserId', // Mock the user's UID here
      },
    },
  }));

describe('QRView Component', () => {
    it('matches snapshot', () => {
        // Mock card data
        const card = {
            "cardId": 992847266036,
            "cardName": "The Ice Cream Card",
            "color": undefined,
            "image": "icecreamRestaurant",
            "loyaltyCard": "loyaltyCards/6xHLagIWsk8Z8Z4ndbO4",
            "loyaltyProgram": "loyaltyPrograms/kAdCUvzu63ybzK09VXEI",
            "progress": 5
        };

        // Render the QRView component with the mock data
        const { toJSON } = render(<QRView card={card} />);

        // Use Jest's expect to match the snapshot
        expect(toJSON()).toMatchSnapshot();
    });
});
