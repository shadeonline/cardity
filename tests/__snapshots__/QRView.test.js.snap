// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRView Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    The Ice Cream Card
  </Text>
</View>
`;
